### PR TITLE
fix(core): use fs/promises instead of promisifying with the util module

### PR DESCRIPTION
**Pull request by Phillip9587**
## Current Behavior

- the task-runners cache (`packages/workspace/src/tasks-runner/cache.ts`) uses `promisify` of the `util` module instead of `fs/promises`
- ensureDir and mkdir are both imported but ensureDir is only a wrapper for mkdir`s recursive option
## Expected Behavior

- the task-runners cache should use fs/promises which is avialable since nodejs 10
- mkdirSync with the recursive option should be used when possible

### CODE REVIEW

File: packages/workspace/src/tasks-runner/cache.ts [hunk 1]
```diff
@@ -106,10 +100,10 @@ export class Cache {
       await Promise.all(
         outputs.map(async (f) => {
           const src = join(this.root, f);
-          if (await existsAsync(src)) {
+          if (existsSync(src)) {
```
Comment (FrozenPandaz) [inline packages/workspace/src/tasks-runner/cache.ts:103]: Why did this change to sync? I know it's pretty fast to check if things exist but it should stay async. Please change all of the other places as well.

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]
Thank you for your contribution!

Please change `existsSync` back to async.

File: packages/workspace/src/tasks-runner/cache.ts (hunk 1 above)
Comment (Phillip9587) [reply packages/workspace/src/tasks-runner/cache.ts:103]: @FrozenPandaz async exists is deprecated and should not be used according to the Node.js Documentation. https://nodejs.org/api/fs.html#fsexistspath-callback

**Review by FrozenPandaz — APPROVED** [review]
Thanks, I updated it to use `pathExists` which is similar to `exists` but uses `access` under the hood.

**Pull request merged**